ui/org.eclipse.pde.core/src/org/eclipse/pde/core/plugin/PluginRegistry.java

```diff
@@ -264,61 +278,85 @@ public static IPluginModelBase[] getExternalModels() {
+	 * @since 3.19
+	 */
+	public static IPluginModelBase findModel(String id, String version, VersionMatchRule matchRule) {
+		return findModels(id, version, matchRule).findFirst().orElse(null);
+	}
+
+	/**
+	 * @see #findModel(String, String, VersionMatchRule)
+	 * @see #findModels(String, String, VersionMatchRule)
 	 * @since 3.6
+	 * @deprecated If no filter is passed use
+	 *             {@link #findModel(String, String, VersionMatchRule)}, else
+	 *             filter the {@code Stream} returned by
+	 *             {@link #findModels(String, String, VersionMatchRule)}.
 	 */
+	@Deprecated(forRemoval = true, since = "3.19 (removal in 2026-09 or later)")
 	public static IPluginModelBase findModel(String id, String version, int match, PluginFilter filter) {
-		return getMax(findModels(id, version, match, filter));
+		VersionMatchRule rule = VersionUtil.matchRuleFromLiteral(match);
+		return findModels(id, version, rule).filter(asPredicate(filter)).findFirst().orElse(null);
 	}
 
 	/**
-	 * Returns all models matching the given id, version, match rule, and optional filter.
+	 * Returns all models matching the given id, version and match rule sorted
+	 * by descending version.
 	 * <p>
-	 * Target (external) plug-ins/fragments with the same ID as workspace counterparts are not
-	 * considered.
+	 * Target (external) plug-ins/fragments with the same ID as workspace
+	 * counterparts are not considered.
 	 * </p>
 	 * <p>
-	 * Returns plug-ins regardless of whether they are checked/enabled or unchecked/disabled
-	 * on the Target Platform preference page.
+	 * Returns plug-ins regardless of whether they are checked/enabled or
+	 * unchecked/disabled on the Target Platform preference page.
 	 * </p>
-	 * @param id symbolic name of a plug-ins to find
-	 * @param version minimum version, or <code>null</code> to only match on symbolic name
-	 * @param match one of {@link IMatchRules#COMPATIBLE}, {@link IMatchRules#EQUIVALENT},
-	 *  {@link IMatchRules#GREATER_OR_EQUAL}, {@link IMatchRules#PERFECT}, or {@link IMatchRules#NONE}
-	 *  when a version is unspecified
-	 * @param filter a plug-in filter or <code>null</code>
+	 *
+	 * @param id
+	 *            symbolic name of a plug-ins to find
+	 * @param version
+	 *            minimum version, or <code>null</code> to only match on
+	 *            symbolic name
+	 * @param matchRule
+	 *            the {@link VersionMatchRule match rule} a plug-in's version
+	 *            must fulfill with respect to the specified reference version
+	 *            in order to be selected. May not be null.
 	 *
 	 * @return a matching models, possibly an empty collection
+	 * @since 3.19
+	 */
+	public static Stream<IPluginModelBase> findModels(String id, String version, VersionMatchRule matchRule) {
+		Version reference = version != null ? Version.valueOf(version) : null;
+		return selectModels(id, version != null ? p -> matchRule.matches(VersionUtil.getVersion(p), reference) : null);
+	}
```

Comment:
Instead of having one overload of `findModel(s)` that accepts a `VersionRange` and one that accepts a version String plus a `VersionMatchRule` we could also just provide one method with a `VersionRange`.
With the new `VersionMatchRule` one would then use
```
findModel("foo.bar", EQUIVALENT.rangeFor(version))
```
instead of 
```
findModel("foo.bar", versionStr, EQUIVALENT)
```
Of course the level of convenience is also influenced from if one has a `Version` object or a version `String`.
Maybe this justifies to keep both so in case one has a Version object the former solution can be used and if one has a version `String` the latter approach is nicer.

I'm torn on this question.

Reply:
Because there is more than just one difference as described above, I think it is fine to preserve this structure.